### PR TITLE
vsce: init at 2.14.0

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/vsce/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  passthru.tests.version = testers.testVersion {
+    package = vsce;
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/microsoft/vscode-vsce";
+    description = "Visual Studio Code Extension Manager";
+    maintainers = with maintainers; [ aaronjheng ];
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vsce/default.nix:22]: Is there a default meta.platform?

File: pkgs/development/tools/vsce/generate.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p node2nix
+
+VSCE_ROOT="$(
+    cd "$(dirname "$0")"
+    pwd
+)"
+pushd $VSCE_ROOT 1>/dev/null
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vsce/generate.sh:8]: Line 4 to 8 can be replaced with this:

```suggestion
cd "$(dirname "$0")"
```

File: pkgs/development/tools/vsce/generate.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+rm -rf gen && mkdir -p gen
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vsce/generate.sh:10]: We don't need to remove the files. Node2nix will overwrite them.

File: pkgs/development/tools/vsce/generate.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+node2nix \
+    --no-copy-node-env \
+    --node-env ../../node-packages/node-env.nix \
+    --input packages.json \
+    --output gen/packages.nix \
+    --composition gen/composition.nix \
+    --strip-optional-dependencies
+
+popd 1>/dev/null
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vsce/generate.sh:20]: We don't need to go back

**Review by SuperSandro2000 — COMMENTED** [review]
Please drop the gen directory.

File: pkgs/development/tools/vsce/default.nix (hunk 1 above)
Comment (aaronjheng) [reply pkgs/development/tools/vsce/default.nix:22]: Default to Node.js' platforms.

**Comment by aaronjheng** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/204111.

**Pull request closed without merging**